### PR TITLE
VMO-4312 Show error on click of Fix Issue in notification bar or when an input field is touched

### PR DESCRIPTION
**Pull request by poojahpatil90**

### CODE REVIEW

File: src/components/common/ValidationMessage.vue [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <template>
   <div
     :id="messageKey"
-    class="validation-message form-group">
+    class="validation-message form-group" @mousedown="onFocusEvent">
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/components/common/ValidationMessage.vue:4]: Done!

File: src/components/common/ValidationMessage.vue [hunk 2]
```diff
@@ -45,10 +45,26 @@ class ValidationMessage extends mixins(Lang) {
   }
 
   get isValid(): boolean {
-    return !this.errorMessage
+    return !(this.errorMessage && this.dirtyStatusForKey(this.messageKey))
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/components/common/ValidationMessage.vue:49]: @safydy-r  Yes, it is a bit confusing. Let me try explaining 
### Error Message        Dirty/Touched           Valid
Exists                       True                           False (We want to show the error)
Not exists                True                           True (We don't want to show error as there is no error message)
Exists                       False                          True (We don't want to show error until the user touches a field)
Not exists                False                          True (No need to show anything)

Hope this table provides clarity on why the condition exists this way. If you still have some confusion, please let me know.

File: src/store/validation/index.ts [hunk 3]
```diff
@@ -35,10 +35,12 @@ export interface IValidationStatus {
 export interface IValidationState {
   // important context for future debug or testing, keys are index like `flow/flowId
   validationStatuses: { [key: string]: IValidationStatus },
+  dirtyStatuses: {[key: string]: boolean}
 }
 
 export const stateFactory = (): IValidationState => ({
   validationStatuses: {} as { [key: string]: IValidationStatus },
+  dirtyStatuses: {} as {[key: string]: boolean},
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/store/validation/index.ts:43]: @safydy-r  Me and Brett both liked the idea of using the word `dirty` here. This is common word used for form validations in most front-end frameworks(like Angular) to represent a field that was modified in some way. 
If you find it very confusing, I would be open to changing it.

File: src/store/validation/index.ts [hunk 4]
```diff
@@ -56,6 +58,7 @@ export const getters: GetterTree<IValidationState, IRootState> = {
     })
     return ajvErrorForMessageKey
   },
+  dirtyStatusForKey: (state) => (key: string): boolean => state.dirtyStatuses[key],
```
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/store/validation/index.ts:58]: I agree!  Made the changes.

File: src/components/common/ValidationMessage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/common/ValidationMessage.vue]: thanks for the table, I think I got confused about `isValid === false` means `We want to show the error`. I thought it should be the opposite, LOL. 😂 😝 sorry.

File: src/store/validation/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/store/validation/index.ts:43]: This is just a subjective feeling from me, if the 02 of you were good, then let's keep it. Thanks

**Review by safydy-r — APPROVED** [review]
Approved, thanks for the updates

File: src/components/common/ValidationMessage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (poojahpatil90) [reply src/components/common/ValidationMessage.vue]: No problem 😂  I had to draw this table while implementing to not get confused too 😝

**Comment by adeablay** [comment]
Hi @poojahpatil90 , I don't know if this has been reported already. But I get this console error when I click on "show or hide block editor" icon of a block. 

```
vue-router.esm.js?8c4f:2065 Uncaught (in promise) NavigationDuplicated: Avoided redundant navigation to current location: "/trees/018d0703-03fe-4bfa-9d49-446b13388988/interaction-designer/edit/block/014dc891-0ad7-4004-906e-76492c2e248e".
    at createRouterError (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:2066:15)
    at createNavigationDuplicatedError (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:2036:15)
    at HTML5History.confirmTransition (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:2329:18)
    at HTML5History.transitionTo (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:2261:8)
    at HTML5History.replace (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:2614:10)
    at eval (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:3040:22)
    at new Promise (<anonymous>)
    at VueRouter.replace (webpack-internal:///./node_modules/vue-router/dist/vue-router.esm.js:3039:12)
    at VueComponent.handleExpandMinimizeBlockEditor (webpack-internal:///./node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/ts-loader/index.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/components/interaction-designer/block/BlockToolbar.vue?vue&type=script&lang=ts&:124:20)
    at click (webpack-internal:///./node_modules/cache-loader/dist/cjs.js?{"cacheDirectory":"node_modules/.cache/vue-loader","cacheIdentifier":"d4777998-vue-loader-template"}!./node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader/lib/index.js?!./src/components/interaction-designer/block/BlockToolbar.vue?vue&type=template&id=752dfa99&:192:66)
```

**Pull request merged**